core/hash/fetch_spec.rb

```diff
@@ -1,40 +1,14 @@
 require File.expand_path('../../../spec_helper', __FILE__)
 require File.expand_path('../fixtures/classes', __FILE__)
+require File.expand_path('../../../shared/hash/key_error', __FILE__)
 
 describe "Hash#fetch" do
-  it "returns the value for key" do
-    { a: 1, b: -1 }.fetch(:b).should == -1
-  end
-
   context "when the key is not found" do
-    it "raises a KeyError" do
-      lambda { {}.fetch(:a)       }.should raise_error(KeyError)
-      lambda { Hash.new(5).fetch(:a)    }.should raise_error(KeyError)
-      lambda { Hash.new { 5 }.fetch(:a) }.should raise_error(KeyError)
-    end
```

Comment:
It would be good to keep these examples which use different kinds of `Hash`.

Reply:
I've added 3 more `it_behaves_like` calls so it can test all the shared specs in each of the Hash creation methods.